code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,464 @@
+		for(var/obj/item/I in frozen_items)
+			dat += "[I.name]<br/>"
+		dat += "<hr/>"
+
+		user << browse(dat, "window=cryoitems")
+
+	else if(href_list["item"])
+		if(!allowed(user))
+			to_chat(user, "<span class='warning'>Access Denied.</span>")
+			return
+		if(!allow_items) return
+
+		if(frozen_items.len == 0)
+			to_chat(user, "<span class='notice'>There is nothing to recover from storage.</span>")
+			return
+
+		var/obj/item/I = input(usr, "Please choose which object to retrieve.","Object recovery",null) as null|anything in frozen_items
+		if(!I)
+			return
+
+		if(!(I in frozen_items))
+			to_chat(user, "<span class='notice'>\The [I] is no longer in storage.</span>")
+			return
+
+		visible_message("<span class='notice'>The console beeps happily as it disgorges \the [I].</span>")
+
+		I.forceMove(get_turf(src))
+		frozen_items -= I
+
+	else if(href_list["allitems"])
+		if(!allowed(user))
+			to_chat(user, "<span class='warning'>Access Denied.</span>")
+			return
+		if(!allow_items) return
+
+		if(frozen_items.len == 0)
+			to_chat(user, "<span class='notice'>There is nothing to recover from storage.</span>")
+			return
+
+		visible_message("<span class='notice'>The console beeps happily as it disgorges the desired objects.</span>")
+
+		for(var/obj/item/I in frozen_items)
+			I.forceMove(get_turf(src))
+			frozen_items -= I
+
+	updateUsrDialog()
+	return
+
+/obj/item/circuitboard/cryopodcontrol
+	name = "Circuit board (Cryogenic Oversight Console)"
+	build_path = "/obj/machinery/computer/cryopod"
+	origin_tech = "programming=1"
+
+//Cryopods themselves.
+/obj/machinery/cryopod
+	name = "cryogenic freezer"
+	desc = "A man-sized pod for entering suspended animation."
+	icon = 'icons/obj/cryogenic2.dmi'
+	icon_state = "cryopod-open"
+	density = TRUE
+	anchored = TRUE
+	state_open = TRUE
+	unsecuring_tool = null
+
+	var/on_store_message = "has entered long-term storage."
+	var/on_store_name = "Cryogenic Oversight"
+
+	// 15 minutes-ish safe period before being despawned.
+	var/time_till_despawn = 15 * 600 // This is reduced by 90% if a player manually enters cryo
+	var/despawn_world_time = null          // Used to keep track of the safe period.
+
+	var/obj/machinery/computer/cryopod/control_computer
+	var/last_no_computer_message = 0
+
+	// These items are preserved when the process() despawn proc occurs.
+	var/list/preserve_items = list(
+		/obj/item/hand_tele,
+		/obj/item/card/id/captains_spare,
+		/obj/item/device/aicard,
+		/obj/item/device/mmi,
+		/obj/item/device/paicard,
+		/obj/item/gun,
+		/obj/item/pinpointer,
+		/obj/item/clothing/shoes/magboots,
+		/obj/item/areaeditor/blueprints,
+		/obj/item/clothing/head/helmet/space,
+		/obj/item/clothing/suit/space,
+		/obj/item/clothing/suit/armor,
+		/obj/item/defibrillator/compact,
+		/obj/item/reagent_containers/hypospray/CMO,
+		/obj/item/clothing/accessory/medal/gold/captain,
+		/obj/item/clothing/gloves/krav_maga,
+		/obj/item/storage/internal,
+		/obj/item/nullrod
```

Comment:
With some paths changing on TG, this may cause issues potentially.

Also, if there are new theft objectives... like pretty sure the Captain's got a jetpack which is a theft item now? Can you double check these.

Reply:
Might be worth finding a better way of enumerating these in the future.